go/arrow/csv/reader.go

```diff
@@ -884,7 +884,7 @@ func (r *Reader) parseBinaryType(field array.Builder, str string) {
 	}
 	decodedVal, err := base64.StdEncoding.DecodeString(str)
 	if err != nil {
-		panic("cannot decode base64 string " + str)
+		r.err = fmt.Errorf("cannot decode base64 string %s", str)
```

Comment:
should probably AppendNull instead of appending the value anyways after getting this error.